server/persistence/mysql/games.go

```diff
@@ -0,0 +1,569 @@
+func populateScores(
+	scoreBlue, scoreRed, scoreGreen,
+	lagScoreBlue, lagScoreRed, lagScoreGreen uint8,
+) (cur, lag map[model.PlayerColor]int) {
+	curScores := make(map[model.PlayerColor]int, 3)
+	lagScores := make(map[model.PlayerColor]int, 3)
+	if scoreBlue > 0 {
+		curScores[model.Blue] = int(scoreBlue)
+		lagScores[model.Blue] = int(lagScoreBlue)
+	}
+	if scoreRed > 0 {
+		curScores[model.Red] = int(scoreRed)
+		lagScores[model.Red] = int(lagScoreRed)
+	}
+	if scoreGreen > 0 {
+		curScores[model.Green] = int(scoreGreen)
+		lagScores[model.Green] = int(lagScoreGreen)
+	}
+
+	return curScores, lagScores
+}
+
+func addInPopulatedColor(
+	curScores, lagScores map[model.PlayerColor]int,
+	pc map[model.PlayerID]model.PlayerColor,
+) {
+	// if we know what color the players are, but we don't have point entries
+	// for those colors in the scores maps, add zeros
+	for _, color := range pc {
+		if _, ok := curScores[color]; !ok {
+			curScores[color] = 0
+		}
+		if _, ok := lagScores[color]; !ok {
+			lagScores[color] = 0
+		}
+	}
+}
+
+func getCribCards(cribCardInt int32) []model.Card {
+	var cribCards []model.Card
+	var cci int8
+	for i := uint(0); i < 4; i++ {
+		cci = int8(cribCardInt >> (8 * i))
+		c, err := model.NewCardFromTinyInt(cci)
+		if err != nil {
+			// If we've errored here, we assume it just means the card isn't set
+			continue
+		}
+		cribCards = append(cribCards, c)
+	}
+	return cribCards
+}
+
+func serializeCribCards(crib []model.Card) int32 {
+	val := int32(0)
+	for i := uint(0); i < 4; i++ {
+		ti := int8(0x34)
+		if int(i) < len(crib) {
+			ti = crib[i].ToTinyInt()
+		}
+		val |= (int32(ti) << (8 * i))
+	}
+	return val
+}
+
+func getBlockingPlayers(ser []byte) (map[model.PlayerID]model.Blocker, error) {
+	blockers := map[model.PlayerID]model.Blocker{}
+
+	err := json.Unmarshal(ser, &blockers)
+	if err != nil {
+		return nil, err
+	}
+
+	return blockers, nil
+}
+
+func serializeBlockingPlayers(input map[model.PlayerID]model.Blocker) ([]byte, error) {
+	return json.Marshal(input)
+}
+
+func getHands(ser []byte) (map[model.PlayerID][]model.Card, error) {
+	hands := map[model.PlayerID][]model.Card{}
+
+	err := json.Unmarshal(ser, &hands)
+	if err != nil {
+		return nil, err
+	}
+
+	return hands, nil
+}
+
+func serializeHands(input map[model.PlayerID][]model.Card) ([]byte, error) {
+	return json.Marshal(input)
+}
+
+func getPeggedCards(ser []byte) ([]model.PeggedCard, error) {
+	peggedCards := []model.PeggedCard{}
+
+	err := json.Unmarshal(ser, &peggedCards)
+	if err != nil {
+		return nil, err
+	}
+
+	return peggedCards, nil
+}
+
+func serializePeggedCards(input []model.PeggedCard) ([]byte, error) {
+	return json.Marshal(input)
+}
+
+func (g *gameService) getPlayersForGame(
+	p1ID, p2ID model.PlayerID,
+	p3ID, p4ID *model.PlayerID,
+) ([]model.Player, error) {
+
+	if len(p1ID) == 0 {
+		return nil, errors.New(`at least one player required`)
+	}
+	if len(p2ID) == 0 {
+		return nil, errors.New(`at least two players required`)
+	}
```

Comment:
First error is misleading - you say at least one player is required when in fact two are required. Can we combine this into one `if`?